core/org.wso2.carbon.utils/src/main/java/org/wso2/carbon/utils/dbcreator/DatabaseCreator.java

```diff
@@ -349,6 +350,24 @@ private void executeSQLScript() throws Exception {
         }
     }
 
+    private String getDatabaseTypeWithVersion(String databaseType) {
+        try {
+            if (databaseType != null && databaseType.equals("mysql")) {
+                if (conn != null && (!conn.isClosed())) {
+                    DatabaseMetaData metaData = conn.getMetaData();
+                    String databaseProductVersion = metaData.getDatabaseProductVersion();
+                    double versionWithMinorNumber = Double.parseDouble(databaseProductVersion.substring(0, 3));
+                    if (versionWithMinorNumber >= 5.7) {
+                        databaseType = databaseType + "-5.7+";
+                    }
+                }
+            }
+        } catch (SQLException | NumberFormatException e) {
+            log.warn("Unable to retrieve version of the database.");

```

Comment:
I don't think it's a good idea to swallow an SQLException. At least get error message from 'e' and append it to warn log.